saleor/product/tests/test_tasks.py

```diff
@@ -128,6 +128,42 @@ def test_update_products_discounted_prices_for_promotion_task_with_rules_id(
     ) == {rule_id}
 
 
+@patch("saleor.product.tasks.PROMOTION_RULE_BATCH_SIZE", 1)
+@patch("saleor.product.tasks.update_discounted_prices_task.delay")
+@patch("saleor.product.utils.variants.fetch_variants_for_promotion_rules")
+def test_update_products_discounted_prices_for_promotion_task_with_empty_reward_value(
+    fetch_variants_for_promotion_rules_mock,
+    update_discounted_prices_task_mock,
+    promotion_list,
+    collection,
+    product_list,
+):
+    # given
+    Promotion.objects.update(start_date=timezone.now() - timedelta(days=1))
+    PromotionRuleVariant = PromotionRule.variants.through
+    PromotionRuleVariant.objects.all().delete()
+
+    collection.products.add(*product_list[1:])
+
+    rule = PromotionRule.objects.first()
+    rule.reward_value = 0
```

Comment:
I suggest to parametrise the test and add case with `reward_value` = None.

Reply:
Added